### PR TITLE
Minor tweaks in UI

### PR DESCRIPTION
**Pull request by rakol-ms**
Signed-off-by: Rajendra Kolli <rakol@microsoft.com>

This PR contains the following tweaks

* Added condition to render links in pods table only when there is content.
* Added a condition to render a link to workload only if it exists in pod metadata under service details. This condition was observed in the failed clusters.
* Corrected title in filters
* Added no results page for services filter. For workloads some work needs to be done since data is split among two components and is filtered independently while No results page should be rendered by `WorkloadsPivot` component.

### CODE REVIEW

**Comment by rakol-ms** [comment]
filter_title_correction1
filter_title_correction2
no_results_page_svc

File: src/WebUI/Pods/PodsTable.tsx [hunk 1]
```diff
@@ -132,16 +132,16 @@ export class PodsTable extends BaseComponent<IPodsTableProperties> {
     }
 
     private static _renderPodWorkload(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1Pod>, pod: V1Pod): JSX.Element {
-        const textToRender = pod.metadata.ownerReferences.length > 0 ? pod.metadata.ownerReferences[0].name : ""; 
-        const itemToRender: React.ReactNode = (
+        const textToRender = pod.metadata.ownerReferences && pod.metadata.ownerReferences.length > 0 ? pod.metadata.ownerReferences[0].name : "";
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsTable.tsx]: please add following also to the conditions
pod && pod.metadata &&

File: src/WebUI/Pods/PodsTable.tsx [hunk 2]
```diff
@@ -132,16 +132,16 @@ export class PodsTable extends BaseComponent<IPodsTableProperties> {
     }
 
     private static _renderPodWorkload(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1Pod>, pod: V1Pod): JSX.Element {
-        const textToRender = pod.metadata.ownerReferences.length > 0 ? pod.metadata.ownerReferences[0].name : ""; 
-        const itemToRender: React.ReactNode = (
+        const textToRender = pod.metadata.ownerReferences && pod.metadata.ownerReferences.length > 0 ? pod.metadata.ownerReferences[0].name : ""; 
+        const itemToRender: React.ReactNode = textToRender.length > 0 ? (
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsTable.tsx]: textToRender.length > 0, replace this with !!textToRender or textToRender

File: src/WebUI/Resources.js [hunk 3]
```diff
@@ -71,4 +71,6 @@ define("Environments/Providers/Kubernetes/Resources", ["require", "exports"], fu
     exports.WorkloadsZeroDataText = "You can also deploy all kinds of workloads like deployment, replicaset, daemonset etc., and track them in here";
     exports.NoPodsForSvcLinkText = "Learn more about mapping a service";
     exports.NoPodsText = "No pods";
+    exports.ServiceText = "Service";
```
Comment (madhuv-msft) [inline src/WebUI/Resources.js]: as per spec this is 'service', small S

Same with workload also

File: src/WebUI/Services/ServicesFilterBar.tsx [hunk 4]
```diff
@@ -13,17 +13,17 @@ import "../Common/KubeSummary.scss";
...
     serviceList: V1ServiceList;
 }
 
-export class ServicesFilterBar extends BaseComponent<IWorkloadsFilterBarProps> {
+export class ServicesFilterBar extends BaseComponent<IServiceFilterBarProps> {
     public render(): React.ReactNode {
         return (<KubeFilterBar filter={this.props.filter}
             pickListPlaceHolder={Resources.TypeText}
-            keywordPlaceHolder={Resources.PivotServiceText}
+            keywordPlaceHolder={Resources.ServiceText}
```
Comment (madhuv-msft) [inline src/WebUI/Services/ServicesFilterBar.tsx]: as mentioned we should have 'service' and not 'Service', with small S

Comment (madhuv-msft) [inline src/WebUI/Services/ServicesFilterBar.tsx]: as mentioned we should have 'service' and not 'Service', with small S

File: src/WebUI/WorkLoads/WorkloadsFilterBar.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ export interface IWorkloadsFilterBarProps extends IVssComponentProperties {
 export class WorkloadsFilterBar extends BaseComponent<IWorkloadsFilterBarProps> {
     public render(): React.ReactNode {
         return (<KubeFilterBar filter={this.props.filter}
-            keywordPlaceHolder={Resources.PivotWorkloadsText}
+            keywordPlaceHolder={Resources.WorkloadText}
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadsFilterBar.tsx]: 'workload' and not 'Workload'

File: src/WebUI/Common/KubeZeroData.tsx [hunk 6]
```diff
@@ -60,6 +61,12 @@ export class KubeZeroData extends BaseComponent<IKubeZeroDataProps> {
         );
     }
 
+    public static _getNoResultsZeroData(): JSX.Element {
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeZeroData.tsx]: for public function _ is not used

**Comment by rakol-ms** [comment]
> > * Added no results page for services filter. For workloads some work needs to be done since data is split among two components and is filtered independently while No results page should be rendered by `WorkloadsPivot` component.
> 
> how are we following up, by some issue or some other way?

@madhuv-msft I am working on it today. If it cannot be completed EOD, I'll file an issue before merging this PR.

**Pull request merged**